File: cpp/libjoynr/common/CallContext.cpp
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2017 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

#include "joynr/CallContext.h"

namespace joynr
{

void CallContext::setPrincipal(const std::string& principal)
{
    this->principal = principal;
    JOYNR_LOG_TRACE(logger(), "setPrincipal '{}'", this->principal);
}

void CallContext::setPrincipal(std::string&& principal)
{
    this->principal = std::move(principal);
    JOYNR_LOG_TRACE(logger(), "setPrincipal '{}'", this->principal);
}

const std::string& CallContext::getPrincipal() const
{
    return principal;
}

void CallContext::invalidate()
{
    JOYNR_LOG_TRACE(logger(), "invalidate");
    principal = std::string();
}

bool CallContext::operator==(const CallContext& other) const
{
    return principal == other.getPrincipal();
}

bool CallContext::operator!=(const CallContext& other) const
{
    return !(*this == other);
}

} // namespace joynr

File: java/javaapi/src/main/java/io/joynr/provider/CallContext.java
/*
 * #%L
 * %%
 * Copyright (C) 2011 - 2017 BMW Car IT GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package io.joynr.provider;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public class CallContext {
    String principal;
    Map<String, Serializable> context;

    public String getPrincipal() {
        return principal;
    }

    public void setPrincipal(String principal) {
        this.principal = principal;
    }

    public Map<String, Serializable> getContext() {
        return (context != null) ? new HashMap<>(context) : null;
    }

    public void setContext(Map<String, Serializable> context) {
        this.context = (context != null) ? new HashMap<>(context) : null;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((context == null) ? 0 : context.hashCode());
        result = prime * result + ((principal == null) ? 0 : principal.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        CallContext other = (CallContext) obj;
        if (context == null) {
            if (other.context != null)
                return false;
        } else if (!context.equals(other.context))
            return false;
        if (principal == null) {
            if (other.principal != null)
                return false;
        } else if (!principal.equals(other.principal))
            return false;
        return true;
    }

}
